internal/service/ec2/subnet_cidr_reservation_test.go

```diff
@@ -0,0 +1,229 @@
+		CheckDestroy: testAccCheckSubnetCidrReservationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testSubnetCidrReservationConfig_Ipv4,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckSubnetCidrReservationExists(resourceName, &res),
+					resource.TestCheckResourceAttr(resourceName, "cidr_block", "10.1.1.16/28"),
+					resource.TestCheckResourceAttr(resourceName, "description", "test"),
+					resource.TestCheckResourceAttr(resourceName, "reservation_type", "prefix"),
+					acctest.CheckResourceAttrAccountID(resourceName, "owner_id"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateIdFunc: testAccSubnetCidrReservationImportStateIdFunc(resourceName),
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccEC2SubnetCidrReservation_Ipv6(t *testing.T) {
+	var res ec2.SubnetCidrReservation
+	resourceName := "aws_ec2_subnet_cidr_reservation.test"
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acctest.PreCheck(t) },
+		ErrorCheck:   acctest.ErrorCheck(t, ec2.EndpointsID),
+		Providers:    acctest.Providers,
+		CheckDestroy: testAccCheckSubnetCidrReservationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testSubnetCidrReservationConfig_Ipv6,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckSubnetCidrReservationExists(resourceName, &res),
+					resource.TestCheckResourceAttr(resourceName, "reservation_type", "prefix"),
+					acctest.CheckResourceAttrAccountID(resourceName, "owner_id"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateIdFunc: testAccSubnetCidrReservationImportStateIdFunc(resourceName),
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccEC2SubnetCidrReservation_disappears(t *testing.T) {
+	var res ec2.SubnetCidrReservation
+	resourceName := "aws_ec2_subnet_cidr_reservation.test"
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acctest.PreCheck(t) },
+		ErrorCheck:   acctest.ErrorCheck(t, ec2.EndpointsID),
+		Providers:    acctest.Providers,
+		CheckDestroy: testAccCheckSubnetCidrReservationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testSubnetCidrReservationConfig_Ipv4,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckSubnetCidrReservationExists(resourceName, &res),
+					testAccCheckSubnetCidrReservationDelete(resourceName),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func testAccCheckSubnetCidrReservationExists(n string, v *ec2.SubnetCidrReservation) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No reservation ID is set")
+		}
+
+		conn := acctest.Provider.Meta().(*conns.AWSClient).EC2Conn
+		resp, err := conn.GetSubnetCidrReservations(&ec2.GetSubnetCidrReservationsInput{
```

Comment:
Can use

```go
FindSubnetCidrReservationById(conn, rs.Primary.ID, rs.Primary.Attributes["subnet_id"])
```

here instead of duplicating the find logic.